src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,135 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_CAPACITY = 10;
+    private Object[] elementData;
+    private int ammountOfElements;
+
+    public ArrayList() {
+        ammountOfElements = 0;
+        this.elementData = (T[]) new Object[DEFAULT_ARRAY_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (ensureCapacity(ammountOfElements + 1)) {
+            elementData[ammountOfElements++] = value;
+        } else {
+            Object[] oldData = elementData;
+            newArrayCapasity(elementData.length);
+            System.arraycopy(oldData, 0, elementData, 0, ammountOfElements + 1);
+            elementData[ammountOfElements++] = value;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > ammountOfElements + 1 || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("ammountOfElements: "
+                    + ammountOfElements + "<" + "index: " + index);
+        }
+        if (ensureCapacity(ammountOfElements + 1)) {
+            System.arraycopy(elementData, index, elementData,
+                    index + 1, ammountOfElements + 1 - index);
+            elementData[index] = value;
+            ammountOfElements++;
+        } else {
+            Object[] oldData = elementData;
+            newArrayCapasity(elementData.length);
+            System.arraycopy(oldData, index, elementData,
+                    index + 1, ammountOfElements + 1 - index);
+            elementData[index] = value;
+            ammountOfElements++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (ensureCapacity(ammountOfElements + list.size())) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        } else {
+            Object[] oldData = elementData;
+            newArrayCapasity(ammountOfElements + list.size());
```

Comment:
you don't need to do ensureCapacity here, you're doing it inside add() method anyway